dataland-community-manager/src/main/kotlin/org/dataland/datalandcommunitymanager/utils/DataRequestLogger.kt

```diff
@@ -122,4 +124,30 @@ class DataRequestLogger {
                 "with new message.",
         )
     }
+
+    /**
+     * Logs an appropriate message when the data request priority is updated
+     */
+    fun logMessageForPatchingRequestPriority(
+        dataRequestId: String,
+        requestPriority: RequestPriority?,
```

Comment:
It depends on the context. Of course, in the patch request, the priority is nullable because you don't want to patch it every time. But here, you explicitly want to log a message whenever the requestPriority is patched, so in here, the priority shouldn't be nullable. In other words: making it non-nullable here will throw an error if ever the function is called with a nullable requestPriority which imo enforces better codeQuality. In the function logging the accessStatus you refer to (logMessageForPatchingRequestStatusOrAccessStatus) also the RequestStatus is logged but only accessStatus is nullable, requestStatus is not.

The other way around: You want to call this loggerFunction only when requestPriority is not null, right? But with the code here, you will probably never find out, if this is actually happening, because it accepts also null values. If you want to include both cases, look at the function "logMessageWhenCrossReferencingIdentifierValueWithDatalandCompanyId" in the same file.

Reply:
Ok, I understand. I made it non-nullable as we only want to log the message when something is changed and this case only occurs when the request priority is not null.